src/bio/smd/smd_device.c

```diff
@@ -349,13 +349,28 @@ smd_dev_replace(uuid_t old_id, uuid_t new_id, d_list_t *pool_list)
 	/* Replace old device ID with new device ID in target table */
 	for (i = 0; i < dev.sd_tgt_cnt; i++) {
 		uint32_t tgt_id = dev.sd_tgts[i];
-
-		rc = smd_db_upsert(TABLE_TGT, &tgt_id, sizeof(tgt_id),
-				   &id, sizeof(id));
-		if (rc) {
-			D_ERROR("Update target %d failed. "DF_RC"\n",
-				tgt_id, DP_RC(rc));
-			goto tx_end;
+		enum smd_dev_type st;
+
+		for (st = SMD_DEV_TYPE_DATA; st < SMD_DEV_TYPE_MAX; st++) {
+			/* Old device bound to that target? */
+			uuid_copy(id.uuid, old_id);
+			rc = smd_db_fetch(TABLE_TGTS[st], &tgt_id, sizeof(tgt_id), &id, sizeof(id));
+			if (rc && !(rc == -DER_NONEXIST && st > SMD_DEV_TYPE_DATA)) {
+				D_CDEBUG(rc != -DER_NONEXIST, DLOG_ERR, DB_MGMT,
+					 "Fetch target %d failed. "DF_RC"\n", tgt_id,
+					 DP_RC(rc));
+				goto tx_end;
+			}
+			if (rc == 0) {
+				uuid_copy(id.uuid, new_id);
+				rc = smd_db_upsert((char *)TABLE_TGTS[st], &tgt_id, sizeof(tgt_id),
+						   &id, sizeof(id));
+				if (rc) {
+					D_ERROR("Update target %d failed. "DF_RC"\n",
+						tgt_id, DP_RC(rc));
+					goto tx_end;
+				}
+			}
```

Comment:
No need to make these changes so far, because we don't support device monitor/hotplug/replace for WAL & meta device yet. (see latest code in vos_on_blob, also see above changes in create_old_blobs() in this PR).

So let's just assume smd_dev_replace() supports only DATA device for this moment.

Reply:
Would you address Yawei's comments here too?

Reply:
We just don't support META or WAL device faulty/replacing yet. Let's just focus on basic I/O functionalities for this moment, we could consider how to support those advanced features later when we have spare cycles.

Reply:
Updated smd_dev_replace() to support DATA devices only